src/events/neo4j.js

```diff
@@ -0,0 +1,386 @@
+        const {
+            query, params, transactionConfig,
+        } = relevantArgs;
+
+        utils.debugLog('User called Neo4j wrapped Session run function');
+
+        try {
+            const startTime = Date.now();
+
+            const dbApiEvent = createNewNeo4jEvent(this, startTime);
+
+            eventInterface.addToMetadata(
+                dbApiEvent,
+                {},
+                { query, param: params, transaction_config: transactionConfig }
+            );
+
+
+            const resultResponse = wrappedFunction.apply(
+                this,
+                [query, params, transactionConfig]
+            );
+
+            const originalSubscribe = resultResponse.subscribe;
+
+            // Override the Result subscribe with patched subscriber
+            // eslint-disable-next-line func-names
+            resultResponse.subscribe = function (observer) {
+                originalSubscribe.call(this, getPatchedSubscribeMethods(
+                    this,
+                    observer,
+                    arguments,
+                    dbApiEvent,
+                    startTime
+                ));
+            };
+
+            return resultResponse;
+        } catch (error) {
+            tracer.addException(error);
+            return wrappedFunction.apply(this, [query, params, transactionConfig]);
+        }
+    };
+}
+
+
+/**
+ * Wraps neo4j.Transaction.run functions with tracing
+ * @param {Function} wrappedFunction The function to wrap from Neo4j
+ * @returns {Function} The wrapped function
+ */
+function neo4jTransactionRunWrapper(wrappedFunction) {
+    return function internalNeo4jTransactionRunWrapper(...args) {
+        const relevantArgs = getArgsFromFunction(...args, this.constructor.name);
+        const {
+            query, params,
+        } = relevantArgs;
+
+        utils.debugLog('User called Neo4j wrapped Transaction run function');
+
+        try {
+            const startTime = Date.now();
+            const dbApiEvent = createNewNeo4jEvent(this, startTime);
+
+            eventInterface.addToMetadata(
+                dbApiEvent,
+                {},
+                { query, param: params }
+            );
+
+            const resultResponse = wrappedFunction.apply(this, [query, params]);
+            const originalSubscribe = resultResponse.subscribe;
+
+            // Override the Result subscribe with patched subscriber
+            // eslint-disable-next-line func-names
+            resultResponse.subscribe = function (observer) {
+                originalSubscribe.call(this, getPatchedSubscribeMethods(
+                    this,
+                    observer,
+                    arguments,
+                    dbApiEvent,
+                    startTime
+                ));
+            };
+
+            return resultResponse;
+        } catch (error) {
+            tracer.addException(error);
+            return wrappedFunction.apply(this, [query, params]);
```

Comment:
I think if originalSubscribe.call will raise an exception we will call 
wrappedFunction.apply twice the first one before the subscribe function and the second inside the catch maybe we need to change the place of the catch scope

Reply:
Fixed by moving the apply out of the try scope, i think that will solve the issue, right?